src/CodeGen_Posix.cpp

```diff
@@ -115,7 +120,7 @@ CodeGen_Posix::Allocation CodeGen_Posix::create_allocation(const std::string &na
             // We used to do the alloca locally and save and restore the
             // stack pointer, but this makes llvm generate streams of
             // spill/reloads.
-            allocation.ptr = create_alloca_at_entry(i32x8, stack_bytes/32, false, name);
+            allocation.ptr = create_alloca_at_entry(i8, stack_bytes, false, name);
```

Comment:
I thought about this, the problem is that we re-use stack allocations, which might have different types depending on their usage. I thought about just allocating the stack allocation of the type of first use, and then later usages would need to cast (as they do now). What do you think?